source/CAS/PGTStorage/Memcached.php

```diff
@@ -0,0 +1,31 @@
+<?php
+class CAS_PGTStorage_Memcached extends CAS_PGTStorage_AbstractStorage
+{
+    var $_cache;
+
+    function __construct($servers, $ncopies = 2, $expiration = 30){
+        $this->_cache = new Memcached();
+        $this->_cache->addServers($servers);
+        $this->_ncopies = $ncopies;
+        $this->_expiration = $expiration;
+    }
+    function init() {
+        parent::init();
+    }
+
+    function write($pgt,$pgt_iou) {
+        for ($i = 0; $i < $this->_ncopies; $i++){
+            $this->_cache->set($pgt_iou.'-'.$i, $pgt, $this->_expiration);
```

Comment:
There's an edge case issue here that probably doesn't matter in practice, but I wanted to note it. The CAS v2 specification recommends supporting PGTs up to 256 characters long [1], but this implementation is bound at 248 due to the limit on memcached key length (250 bytes).

[1] https://github.com/apereo/cas/blob/master/docs/cas-server-documentation/protocol/CAS-Protocol-V2-Specification.md#331-proxy-granting-ticket-properties

Reply:
Well, if var $pgt_iou can be greather than 250 then the replication technique used would be useless, because the index of the copy is appended to the end. Can be solved using sha256 as suggested